File: Add Code Here/PYTHON/weather.py
import requests


def get_weather(city: str):
    # Enter your API key here
    api_key = "Your_API_Key"
    url = "https://api.openweathermap.org/data/2.5/weather?q=" + \
        city + "&appid=" + api_key + "&units=metric"

    response = requests.get(url).json()

    if response["cod"] == "404":
        print('City not found.')
    else:
        weather = response["main"]
        description = response["weather"][0]["description"]
        print("Location: " + city.capitalize())
        print(description.capitalize())
        print("Temperature: " + str(weather["temp"]) + " C" + "\u00B0")
        print("Feel like: " + str(weather["feels_like"]) + " C" + "\u00B0")
        print("Humidity: " + str(weather["humidity"]) + " %")


city = input("Enter your city: ")
get_weather(city)

File: Add Code Here/JAVASCRIPT/weather.js
const http = require("http");
const fs = require("fs");
var requests = require("requests");


const homeFile = fs.readFileSync("home.html" , "utf-8");
const replaceVal = (tempVal, orgVal) =>{
    let temperature = tempVal.replace("{%tempval%}", orgVal.main.temp);
     temperature = temperature.replace("{%tempmin%}", orgVal.main.temp_min);
     temperature = temperature.replace("{%tempmax%}", orgVal.main.temp_max);
     temperature = temperature.replace("{%location%}", orgVal.name);
     temperature = temperature.replace("{%Country%}", orgVal.sys.country);
     temperature = temperature.replace("{%tempstatus%}", orgVal.weather[0].main);
     return temperature;
   
}

const server = http.createServer((req,res) => {
    if(req.url =="/"){
        requests("http://api.openweathermap.org/data/2.5/weather?q=Ghaziabad&appid=5550d112cd1c40c065d69df3fbf1337b")
        .on("data",  (chunk)=>{
            const objdata = JSON.parse(chunk);
            const arrData = [objdata];
            const realTimeData = arrData
            .map((val) => replaceVal(homeFile, val)).join("");
                res.write(realTimeData);
            
        })
        .on("end" , (err) => {
            if (err) return console.log("Connection closed due to error");
            res.end();
        });
    }

});

server.listen(8000, "127.0.0.1");
